src/libYARP_os/src/yarp/os/Log.cpp

```diff
@@ -281,6 +281,51 @@ inline const char* logTypeToBgColor(yarp::os::Log::LogType t)
     }
 }
 
+inline const char* compNameToColor(const char* comp_name)
+{
+    if (!comp_name || comp_name[0] == '\0' || !yarp::os::impl::LogPrivate::colored_output.load()) {
+        return "";
+    }
+
+    // Hashing the component for every log line is probably not optimal, but in
+    // using a hash table will have the hash + a search, and since the
+    // LogCallback has only the name of the component, we cannot store the color
+    // in the LogComponent without breaking the API, therefore this is the less
+    // invasive option.
+    // Anyway, this is enabled only when YARP_COLORED_OUTPUT is set, therefore
+    // it can be easily disabled if this slows down the execution.
+    static std::hash<std::string> hsh; // FIXME C++17: Use string_view
+    std::size_t comp_hash = hsh(comp_name) % 12;
+    switch (comp_hash) {
+    case 0:
+        return RED;
```

Comment:
I know that it is not part of this PR but isn't better to use `constexpr char[]` instead of defines?